### ISSUE TITLE
False positive consider-using-dict-comprehension

### ISSUE THREAD

**Issue by sam-s** [PyCQA/pylint#2963]
### Steps to reproduce
run `pylint` on
```
dict([s.split("=") for s in ["a=b","c=d"]])
```

### Current behavior
```
refactor (R1717, consider-using-dict-comprehension, ) Consider using a dictionary comprehension
```

### Expected behavior

no diagnostics because the explicit `dict([])` _cannot_ be replaced with dictionary comprehension because there is no colon `:` in the pre-`for` iteration clause

### pylint --version output
```
pylint 2.3.1
astroid 2.2.5
Python 3.7.3 (default, Mar 27 2019, 09:23:15) 
[Clang 10.0.1 (clang-1001.0.46.3)]
```

**Comment by PCManticore** [comment]
Thanks, I agree this didn't make sense.
